lib/SearchAndSort/components/DateRangeFilter/DateRangeFilter.js

```diff
@@ -170,10 +170,10 @@ const TheComponent = ({
 
   React.useEffect(() => {
     setFilterValue({
-      ...getInitialValidationData(selectedValues, dateFormat),
+      ...getInitialValidationData(selectedValues, dateFormat, requiredFields),
       selectedValues
     });
-  }, [selectedValues, dateFormat]);
+  }, [selectedValues, dateFormat, requiredFields]);
```

Comment:
Let's put requiredFields.start and requiredFields.end into deps.

Reply:
This is an array, so we can just add itself

Reply:
We can also destructure the array.
```
const [startDate, endDate] = requiredFields;

useEffect(() => {
  ...
}, [selectedValues, dateFormat, startDate, endDate]);
```

Reply:
In this case we will have to pass `startDate` and `endDate` as a custom array like this too:
```
useEffect(() => {
  setFilterValue({
      ...getInitialValidationData(selectedValues, dateFormat, [startDate, endDate]),
      selectedValues
    });
}, [selectedValues, dateFormat, startDate, endDate]);
```

I don't like passing it like this because we already have a `requiredFields` prop that we can use, and desctructuring it, giving new names to items and then passing the same array in this way makes it too confusing